src/test/java/org/mockitousage/junitrule/StubbingWarningsJUnitRuleTest.java

```diff
@@ -1,242 +1,191 @@
+                    "[MockitoHint]  ...args ok? -> at org.mockitousage.junitrule.StubbingWarningsJUnitRuleTest.useStubbingWithArg(StubbingWarningsJUnitRuleTest.java:0)\n" +
+                    "[MockitoHint] 2. Unused... -> at org.mockitousage.junitrule.StubbingWarningsJUnitRuleTest.declareStubbingWithArg(StubbingWarningsJUnitRuleTest.java:0)\n" +
+                    "[MockitoHint]  ...args ok? -> at org.mockitousage.junitrule.StubbingWarningsJUnitRuleTest.useStubbingWithArg(StubbingWarningsJUnitRuleTest.java:0)\n" +
+                    "[MockitoHint]  ...args ok? -> at org.mockitousage.junitrule.StubbingWarningsJUnitRuleTest.useStubbingWithArg(StubbingWarningsJUnitRuleTest.java:0)\n",
+                        filterLineNo(logger.getLoggedInfo()));
+            }
+        });
+
+        //when
+        declareStubbingWithArg(mock, "a");
+        declareStubbingWithArg(mock, "b");
+
+        useStubbingWithArg(mock, "c");
+        useStubbingWithArg(mock, "d");
+
+        throw new AssertionError("x");
     }
 
     @Test
     public void reports_only_mismatching_stubs() throws Throwable {
-        try {
-            //when
-            jUnitRule.apply(new Statement() {
-                public void evaluate() throws Throwable {
-                    IMethods mock = mock(IMethods.class);
-
-                    declareStubbingWithArg(mock, "a"); // <-- used
-                    declareStubbingWithArg(mock, "b"); // <-- unused
-
-                    useStubbingWithArg(mock, "a");
-                    useStubbingWithArg(mock, "d"); // <-- arg mismatch
-
-                    throw new AssertionError("x");
-                }
-            }, dummy, new DummyTestCase()).evaluate();
-
-            //then
-            fail();
-        } catch (AssertionError e) {
-            assertEquals("x", e.getMessage());
+        //expect
+        rule.expectFailure(new SafeJUnitRule.FailureAssert() {
+            public void doAssert(Throwable t) {
+            assertEquals("x", t.getMessage());
             assertEquals(
-                "[MockitoHint] DummyTestCase.null (see javadoc for MockitoHint):\n" +
+                "[MockitoHint] StubbingWarningsJUnitRuleTest.reports_only_mismatching_stubs (see javadoc for MockitoHint):\n" +
                 "[MockitoHint] 1. Unused... -> at org.mockitousage.junitrule.StubbingWarningsJUnitRuleTest.declareStubbingWithArg(StubbingWarningsJUnitRuleTest.java:0)\n" +
                 "[MockitoHint]  ...args ok? -> at org.mockitousage.junitrule.StubbingWarningsJUnitRuleTest.useStubbingWithArg(StubbingWarningsJUnitRuleTest.java:0)\n",
                     filterLineNo(logger.getLoggedInfo()));
-        }
+            }
+        });
+
+        //when
+        declareStubbingWithArg(mock, "a"); // <-- used
+        declareStubbingWithArg(mock, "b"); // <-- unused
+
+        useStubbingWithArg(mock, "a");
+        useStubbingWithArg(mock, "d"); // <-- arg mismatch
+
+        throw new AssertionError("x");
```

Comment:
Could be a method in `SafeJUnitRule` ? like `triggerFailure()`, this could avoid to test ourselves `assertEquals("x", t.getMessage());`, `SafeJUnitRule` would do that before running the custom `FailureAssert`.

And this would make explicit that the assertion error is related to the rule.